package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -64,6 +87,69 @@ def save(self):
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
-        session_store_data = RunModel(id=self._session_id, blob=self.to_json())
+
+        session_store_data = RunModel(id=self._session_id, blob=self._to_json())
         database.add(session_store_data)
         database.commit()
+        self.sync()
+
+    def _upload(self):
+        """Upload the session store db to s3"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        db_name = _get_dbname()
+        fs = fsspec.filesystem(protocol)
+        with open(self.location, "rb") as file:
+            with fs.open(f"{self._remote_path}/{db_name}", "wb") as s3f:
+                s3f.write(file.read())
+
+    def _download(self) -> List[str]:
+        """Download all the dbs from an s3 bucket"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        fs = fsspec.filesystem(protocol)
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        for database in databases:
+            database_name = os.path.basename(database)
+            with fs.open(database, "rb") as file:
+                db_data = file.read()
+            db_loc = f"{self._path}/{database_name}"
+            with open(db_loc, "wb") as temp_db:
+                temp_db.write(db_data)
+            databases_location.append(db_loc)
+        return databases_location
+
+    def _merge(self, databases_location: List[str]):
```

Comment:
I find this method quite hard to read. Adding comments would probably make it easier to follow.